### PR TITLE
Update bootstrap process to skip key generation

### PR DESCRIPTION
**Pull request by danuio**
Resolves https://github.com/dapperlabs/flow-go/issues/4960.

This PR removes the Staking and Network key generation of internal nodes from the `finalize` bootstrap command. We now run the `keygen` command prior to this to generate the internal node keys.

This also PR adds a new flag to `finalize` command which reads in the private key data generated by the `keygen` command. `--internal-priv-dir`

```console
./bootstrap finalize \
    --config ./mainnet1/node-config.json \
    --internal-priv-dir <directory_generated_from_keygen_cmd>
    --partner-dir ./mainnet1/partner-node-infos/ \
    --partner-stakes ./mainnet1/partner-stakes.json \
    -o ./genesis-infos-mainnet1 \
    --collection-clusters 4 \
    --fast-kg \
    --root-commit 6e1adf15689eaf5ea6859bcdd0b510f5eb4c34dac878d8577b3f65bc20c3f312 \
    --root-chain main \
    --root-height 7601063 \
    --root-parent ab1ee18b6e1c0ee11cc021c26a17c694c627699a576e85f7013cd743bdbc7877 \
    --epoch-counter 0
```

### CODE REVIEW

**Review by tarakby — APPROVED** [review]

File: cmd/bootstrap/cmd/finalize.go [hunk 1]
```diff
@@ -194,6 +205,35 @@ func assemblePartnerNodes() []model.NodeInfo {
 	return nodes
 }
 
+func assembleInternalNodes() []model.NodeInfo {
+
+	privInternals := readInternalNodes()
+	log.Info().Msgf("read %v internal private node-info files", len(privInternals))
+
+	stakes := internalStakesByAddress()
+	log.Info().Msgf("read %v stakes for partner nodes", len(stakes))
```
Comment (danuio) [inline cmd/bootstrap/cmd/finalize.go:214]:
```suggestion
	log.Info().Msgf("read %v stakes for internal nodes", len(stakes))
```

File: cmd/bootstrap/cmd/finalize.go [hunk 2]
```diff
@@ -194,6 +205,35 @@ func assemblePartnerNodes() []model.NodeInfo {
...
+
+	var nodes []model.NodeInfo
+	for _, internal := range privInternals {
+		// validate every single internal node
+		nodeID := validateNodeID(internal.NodeID)
+		stake := validateStake(stakes[internal.Address])
+
```
Comment (zhangchiqing) [inline cmd/bootstrap/cmd/finalize.go:221]: Could we also validate the node address with `validateAddressFormat`

File: cmd/bootstrap/cmd/finalize.go [hunk 3]
```diff
@@ -242,6 +282,47 @@ func readPartnerNodes() []model.PartnerNodeInfoPub {
...
+func internalStakesByAddress() map[string]uint64 {
+	// read json
+	var configs []model.NodeConfig
+	readJSON(flagConfig, &configs)
+	log.Info().Msgf("read internal %v node configurations", configs)
+
+	stakes := make(map[string]uint64)
+	for _, config := range configs {
+		stakes[config.Address] = config.Stake
```
Comment (zhangchiqing) [inline cmd/bootstrap/cmd/finalize.go:320]: let's check if the address has duplication, can report error if there is

File: cmd/bootstrap/cmd/finalize.go [hunk 4]
```diff
@@ -263,3 +331,40 @@ func mergeNodeInfos(internalNodes, partnerNodes []model.NodeInfo) []model.NodeIn
 
 	return nodes
 }
+
+// Validation utilility methods ------------------------------------------------
```
Comment (danuio) [inline cmd/bootstrap/cmd/finalize.go:335]: utility
```suggestion
// Validation utility methods ------------------------------------------------
```

File: cmd/bootstrap/cmd/finalize.go [hunk 5]
```diff
@@ -263,3 +331,40 @@ func mergeNodeInfos(internalNodes, partnerNodes []model.NodeInfo) []model.NodeIn
...
+
+func validateNetworkPubKey(key encodable.NetworkPubKey) encodable.NetworkPubKey {
+	if key.PublicKey == nil {
+		log.Fatal().Msg("NetworkPubKey must not be nil")
+	}
+	return key
+}
+
+func validateStakingPubKey(key encodable.StakingPubKey) encodable.StakingPubKey {
+	if key.PublicKey == nil {
```
Comment (zhangchiqing) [inline cmd/bootstrap/cmd/finalize.go:352]: This validation is insufficient. Have to use [validateAddressFormat](https://github.com/onflow/flow-go/blob/e7d87c2a8e2fc3e1ef6613729c7ee7ff51467810/cmd/bootstrap/cmd/key.go#L104)

File: cmd/bootstrap/cmd/finalize.go [hunk 6]
```diff
@@ -263,3 +331,40 @@ func mergeNodeInfos(internalNodes, partnerNodes []model.NodeInfo) []model.NodeIn
...
+		log.Fatal().Msg("StakingPubKey must not be nil")
+	}
+	return key
+}
+
+func validateAddress(address string) string {
+	if address == "" {
```
Comment (zhangchiqing) [inline cmd/bootstrap/cmd/finalize.go:359]: This validation is insufficient. Have to use validateAddressFormat

Comment (danuio) [reply cmd/bootstrap/cmd/finalize.go:359]: Oh my bad didn't see this.

**Review by zhangchiqing — APPROVED** [review]
Looks good

**Review by vishalchangrani — APPROVED** [review]
lgtm

**Pull request merged**